sound/soc/sof/intel/hda.c

```diff
@@ -987,7 +989,7 @@ static int hda_generic_machine_select(struct snd_sof_dev *sdev)
 			dev_info(bus->dev, "using HDA machine driver %s now\n",
 				 hda_mach->drv_name);
 
-			if (codec_num == 1)
+			if (codec_num == 1 && HDA_IDISP_CODEC(bus->codec_mask))
```

Comment:
minor again, not following here why codec count is important. previous author should have commented this.